plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -167,10 +167,13 @@ func findNodesThatFit(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.No
 
 	if len(filtered) > 0 && len(extenders) != 0 {
 		for _, extender := range extenders {
-			filteredList, err := extender.Filter(pod, &api.NodeList{Items: filtered})
+			filteredList, failedMap, err := extender.Filter(pod, &api.NodeList{Items: filtered})
 			if err != nil {
 				return api.NodeList{}, FailedPredicateMap{}, err
 			}
+			for failedNodeName, failedMsg := range failedMap {
+				failedPredicateMap[failedNodeName] = failedMsg
```

Comment:
Ideally failedPredicateMap should be failedNodeMap as a node can fail because of a predicate or an extender.